tools/cell_census_builder/tests/test_util.py

```diff
@@ -0,0 +1,133 @@
+import numpy as np
+import pytest
+from scipy.sparse import coo_matrix, csr_matrix, triu
+
+from tools.cell_census_builder.util import array_chunker, is_nonnegative_integral, uricat
+
+
+def test_is_nonnegative_integral() -> None:
+    X = np.array([1, 2, 3], dtype=np.float32)
+    assert is_nonnegative_integral(X)
+
+    X = np.array([-1, 2, 3], dtype=np.float32)
+    assert not is_nonnegative_integral(X)
+
+    X = np.array([[1, 2, 3], [4, 5, 6]], dtype=np.float32)
+    assert is_nonnegative_integral(X)
+
+    X = np.array([[-1, 2, 3], [4, 5, 6]], dtype=np.float32)
+    assert not is_nonnegative_integral(X)
+
+    X = np.array([[1.2, 0, 3], [4, 5, 6]], dtype=np.float32)
+    assert not is_nonnegative_integral(X)
+
+    X = np.zeros((3, 4), dtype=np.float32)
+    assert is_nonnegative_integral(X)
+
+    X = csr_matrix([[1, 2, 3], [4, 5, 6]])
+    assert is_nonnegative_integral(X)
+
+    X = csr_matrix([[-1, 2, 3], [4, 5, 6]])
```

Comment:
these are also int64 - same issue as with ndarray, they need a float dtype to be specified.

Reply:
Apologies, for some reason I thought scipy.sparse created float matrices by default. Fixing it.